### PR TITLE
feat: migrate user tasks

### PR DESCRIPTION
**Pull request by loriepisicchio**
## Issue

https://gravitee.atlassian.net/browse/APIM-1383

## Description

Migrate user tasks page 

screenshot-localhost_3000-2023 04 18-10_48_39

Empty tasks : 
screenshot-localhost_3000-2023 04 18-10_47_49
 
I've taken the opportunity to add a loader while data are fetched
screenshot-localhost_3000-2023 04 18-10_48_04

🏗️ Your changes can be tested here and will be available soon:
      Console: [https://3699.team-apim.gravitee.dev/console](https://3699.team-apim.gravitee.dev/console)
      Portal: [https://3699.team-apim.gravitee.dev/portal](https://3699.team-apim.gravitee.dev/portal)
      Management-api: [https://3699.team-apim.gravitee.dev/api/management](https://3699.team-apim.gravitee.dev/api/management)
      Gateway v4: [https://3699.team-apim.gravitee.dev](https://3699.team-apim.gravitee.dev)
      Gateway v3: [https://3699.gateway-v3.team-apim.gravitee.dev](https://3699.gateway-v3.team-apim.gravitee.dev)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-bmhahygcak.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/tasks/tasks.component.ts [hunk 1]
```diff
@@ -13,42 +13,83 @@
...
-  private tasks: PagedResult<Task>;
+class TaskData {
+  icon: string;
+  title: string;
+  message: string;
+  details: string;
+  action: string;
+  createdAt: number;
+  type: string;
+  data: any;
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/tasks/tasks.component.ts]: Could we try with :
```suggestion
  data: unknown; 
```

File: gravitee-apim-console-webui/src/management/tasks/tasks.component.ts [hunk 2]
```diff
@@ -13,42 +13,83 @@
...
+  tasks: PagedResult<Task>;
+  data: TaskData[];
+  loading = false;
 
-  /* @ngInject */
   constructor(
-    private readonly $state: StateService,
-    private readonly UserService: UserService,
-    private readonly ngTaskService: TaskService,
+    private readonly tasksService: TaskService,
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/tasks/tasks.component.ts]: tasksService or taskService ? what it the better ? do we need to rename the service ?

File: gravitee-apim-console-webui/src/management/tasks/tasks.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+  tasks.populate(responseData);
+
+  let fixture: ComponentFixture<TasksComponent>;
+  let harness: TasksHarness;
+  const init = async (tasks: PagedResult<Task>) => {
+    await TestBed.configureTestingModule({
+      declarations: [TasksComponent],
+      imports: [NoopAnimationsModule, TasksModule, MatIconTestingModule],
+      providers: [
+        { provide: TaskService, useValue: { getTasks: () => of(tasks) } },
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/tasks/tasks.component.spec.ts]: hann I really prefer to use httpTesting it often reveals bad rxjs behavior that can be hidden by this kind of mock.
(I'm not asking for a change. Just a thought)

File: gravitee-apim-console-webui/src/management/tasks/tasks.component.scss [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        margin-right: 8px;
+      }
+    }
+  }
+
+  &__content {
+    &__details {
+      margin-top: 2px;
+      font-size: 12px;
+      color: hsl(0, 0%, 45%);
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/tasks/tasks.component.scss]: can we use a color form ui-P ?

File: gravitee-apim-console-webui/src/management/tasks/tasks.component.ts [hunk 5]
```diff
@@ -70,78 +111,136 @@ class TasksComponentController implements IController, IOnInit, IOnDestroy {
...
+      <code>${task.data.apiName}</code> from environment <strong>${task.data.sourceEnvironmentName}</strong> to
+      environment <strong>${task.data.targetEnvironmentName}</strong>`;
       default:
-        return 'Unknown task';
+        return '';
     }
   }
 
-  title(task: Task): string {
+  getDetails(task: Task): string {
```
Comment (ThibaudAV) [inline gravitee-apim-console-webui/src/management/tasks/tasks.component.ts]: we could pass them to private ? or external to the class like `const getDetails = (task: Task): string => {...}`

**Review by ThibaudAV — APPROVED** [review]
I wonder if we could add the page component in a tab of the new home? 😈 

some little feedback but it LGTM

**Comment by loriepisicchio** [comment]
Thanks for the feedback, I applied the changes.
And also added tasks to a new tab in the home page
screenshot-localhost_3000-2023 04 24-09_29_56

File: gravitee-apim-console-webui/src/management/tasks/tasks.component.spec.ts (hunk 3 above)
Comment (loriepisicchio) [reply gravitee-apim-console-webui/src/management/tasks/tasks.component.spec.ts]: You're right, I changed it :-)

**Comment by ThibaudAV** [comment]
it's @gaetanmaisse who will be happy 😁

**Review by ThibaudAV — APPROVED** [review]

**Comment by gaetanmaisse** [comment]
It's incredible! 🚀

**Pull request merged**